src/imagery/i.eodag/i.eodag.py

```diff
@@ -0,0 +1,326 @@
+# % answer: S1_SAR_GRD
+# % guisection: Filter
+# %end
+
+# %option G_OPT_M_DIR
+# % key: output
+# % description: Name for output directory where to store downloaded data OR search results
+# % required: no
+# % guisection: Output
+# %end
+
+# %option
+# % key: id
+# % type: string
+# % multiple: yes
+# % description: List of scenes IDs to download
+# % guisection: Filter
+# %end
+
+# %option
+# % key: provider
+# % type: string
+# % description: Provider to use for searching/downloading, if none is provided the searching will be done according to the config file priority
+# % required: no
+# % guisection: Filter
+# %end
+
+# %option
+# % key: start
+# % type: string
+# % description: Start date (in any ISO 8601 format), by default it is 60 days ago
+# % guisection: Filter
+# %end
+
+# %option
+# % key: end
+# % type: string
+# % description: End date (in any ISO 8601 format)
+# % guisection: Filter
+# %end
+
+# %option
+# % key: relation
+# % type: string
+# % description: Relation with area of interest
+# % options: intersects, contains, within
+# % answer: intersects
+# % guisection: Optional
+# %end
+
+# %option
+# % key: clouds
+# % type: integer
+# % description: Maximum cloud cover percentage for Landsat scene
+# % required: no
+# % guisection: Filter
+# %end
+
+# %option
+# % key: timeout
+# % type: integer
+# % description: Download timeout in seconds
+# % answer: 300
+# % guisection: Optional
+# %end
+
+
+# %flag
+# % key: l
+# % description: List the search result without downloading
+# %end
+
+# %flag
+# % key: e
+# % description: Extract the downloaded the datasets
+# %end
+
+# %flag
+# % key: d
+# % description: Delete the product archieve after downloading
+# %end
+
+
+import sys
+import os
+import getpass
+from datetime import *
+import grass.script as gs
+from grass.exceptions import ParameterError
+
+
+def create_dir(dir):
+    if not os.path.isdir(dir):
+        try:
+            os.makedirs(dir)
+            return 0
+        except Exception as e:
+            gs.warning(_("Could not create directory {}").format(dir))
+            return 1
+    gs.verbose(_("Directory {} already exists").format(dir))
+    return 0
+
+
+def get_bb(vector=None):
+    args = {}
+    if vector:
+        args["vector"] = vector
+    # are we in LatLong location?
+    kv = gs.parse_command("g.proj", flags="j")
+    if "+proj" not in kv:
+        gs.fatal("Unable to get bounding box: unprojected location not supported")
+    if kv["+proj"] != "longlat":
+        info = gs.parse_command("g.region", flags="uplg", **args)
+        return {
+            "lonmin": info["nw_long"],
+            "latmin": info["sw_lat"],
+            "lonmax": info["ne_long"],
+            "latmax": info["nw_lat"],
+        }
+    info = gs.parse_command("g.region", flags="upg", **args)
+    return {
+        "lonmin": info["w"],
+        "latmin": info["s"],
+        "lonmax": info["e"],
+        "latmax": info["n"],
+    }
+
+
+def download_products(products_list):
```

Comment:
A function with just one line of code just increases the number of lines of code. But it can be OK if you intend to extend it later on...